tests/test_mambabuild.py

```diff
@@ -1,28 +1,23 @@
 import pytest
-import pathlib
+from pathlib import Path
 from subprocess import check_call, CalledProcessError
 
+recipes_dir = Path(__file__).parent / "recipes"
 
-def test_build_recipes():
-    recipes_dir = pathlib.Path(__file__).parent / "recipes"
+recipes = [str(x) for x in recipes_dir.iterdir() if x.is_dir()]
+notest_recipes = [x for x in recipes if Path(x).name in ["baddeps"]]
 
-    recipes = [str(x) for x in recipes_dir.iterdir() if x.is_dir()]
 
+@pytest.mark.parametrize("recipe", recipes)
+def test_build_recipes(recipe):
     expected_fail_recipes = ["baddeps"]
-    for recipe in recipes:
-        recipe_name = pathlib.Path(recipe).name
-        print(f"Running {recipe_name}")
-        if recipe_name in expected_fail_recipes:
-            with pytest.raises(CalledProcessError):
-                check_call(["conda", "mambabuild", recipe])
-        else:
+    if Path(recipe).name in expected_fail_recipes:
+        with pytest.raises(CalledProcessError):
             check_call(["conda", "mambabuild", recipe])
+    else:
+        check_call(["conda", "mambabuild", recipe])
 
 
-def test_build_notest():
-    recipes_dir = pathlib.Path(__file__).parent / "recipes"
-
-    recipes = [str(x) for x in recipes_dir.iterdir() if x.is_dir()]
-    recipe = recipes[0]
```

Comment:
This was effectively `str(recipes_dir / "baddeps")`. I extracted that into `notest_recipes` to make it explicit.